### PR TITLE
adds QGB querier interfaces 

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview
Closes https://github.com/celestiaorg/orchestrator-relayer/issues/41

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: p2p/querier.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		ctx context.Context,
+		nonce uint64,
+	) ([]types.DataCommitmentConfirm, error)
+
+	// QueryValsetConfirms get all the valset confirms in store for a certain nonce.
+	QueryValsetConfirms(
+		ctx context.Context,
+		nonce uint64,
+	) ([]types.ValsetConfirm, error)
+}
```
Comment (evan-forbes) [inline p2p/querier.go:62]: I'm not sure if this is a good starting point for the interface we're looking for imho. For example, We are not planning on having multiple implementations of `QueryForTwoThirds`, so I'm not sure atm if it makes sense to have it. It seems like that, and many of the others here, can simply be a function or a method.

File: rpc/tm_querier.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// TmQuerierI queries tendermint for commitments and events.
+type TmQuerierI interface {
+	Stop() error
+	// QueryCommitment queries tendermint for a commitment for the set of blocks
+	// defined by `beginBlock` and `endBlock`.
+	QueryCommitment(ctx context.Context, beginBlock uint64, endBlock uint64) (bytes.HexBytes, error)
+
+	// SubscribeEvents subscribe to the events named `eventName`.
+	SubscribeEvents(ctx context.Context, subscriptionName string, eventName string) (<-chan coretypes.ResultEvent, error)
+}
```
Comment (evan-forbes) [inline rpc/tm_querier.go:19]: do you think we should add proofs to this?

File: p2p/querier.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	"time"
+
+	celestiatypes "github.com/celestiaorg/celestia-app/x/qgb/types"
+	"github.com/celestiaorg/orchestrator-relayer/types"
+)
+
+// QuerierI queries the P2P network for confirms.
+// Note: for now, we will be querying the P2P network directly without storing anything in the store.
+// this would make it easier to have a simple working version. Then, we can iterate from there on.
+type QuerierI interface {
```
Comment (evan-forbes) [inline p2p/querier.go:14]: [not blocking nit]
The I at the end feels redundant and is usually not added unless there is something else that using the same name. If that's the case, then imho I think we should change one of the names so they don't match

**Review by evan-forbes — COMMENTED** [review]
left one blocking question/comment on the QuerierI interface, but otherwise looks :+1:

File: p2p/querier.go (hunk 1 above)
Comment (SweeXordious) [reply p2p/querier.go:62]: > If these things are just for testing purposes

Yes, 100% they are. As stated here https://github.com/celestiaorg/orchestrator-relayer/pull/50#discussion_r1060298133, I would love to see some examples of mocks to improve these big interfaces.

File: rpc/tm_querier.go (hunk 2 above)
Comment (SweeXordious) [reply rpc/tm_querier.go:19]: I'm not sure where to put the proof generation code. I was thinking of adding it to V1, but we will be deploying a mocked version of it, should it belong to V1-mocked? 

The thing is, the versions have different structures. Thus, it won't be as easy as back porting the PR to the version we want.  

What do you think? Where should the command generating the proofs and verifying them be?

File: p2p/querier.go (hunk 3 above)
Comment (SweeXordious) [reply p2p/querier.go:14]: There will be the implementation of `QuerierI` called `Querier`, unless we find  a better name for it :cat: . I agree, the current naming scheme I have been following for these PRs is a bit dull

File: p2p/querier.go (hunk 1 above)
Comment (evan-forbes) [reply p2p/querier.go:62]: I think we should just not mock anything, and use the testnode code, which should be pretty easy to setup a testsuite for, and can have super short block times. If we really wanted to get clever, we can store some state/blocks as test data, and then boot up that "frozen" chain. 

Alternatively, we could generate mocks for all of the core/app rpc endpoints needed, but that's only marginally better than the other form of mocks and not be worth the effort

File: p2p/querier.go (hunk 3 above)
Comment (evan-forbes) [reply p2p/querier.go:14]: okay, yeah its not the end of the world lol

on a unrelated note, why is the querier in the p2p package?

Comment (SweeXordious) [reply p2p/querier.go:14]: there are multiple queriers, the tendermint querier (for getting the data commitments, the roots etc from tendermint) and the qgb module querier (for getting the attestations from the state machine) and the p2p querier (for eventually getting the attestations confirms)

**Review by evan-forbes — APPROVED** [review]
sure, sounds good

**Pull request merged**